### PR TITLE
Adding extension flag in jam pack as a fallback

### PR DESCRIPTION
**Pull request by pacostas**
## Summary

This PR supports using the --extension flag when calling `jam pack` . At the moment the --extension option is not supported by jam although there is a draft PR for supporting it here: https://github.com/paketo-buildpacks/jam/pull/234

## Use Cases

In case of  buildpack.toml file is not available, instead of using --buildpack flag to package a buildpack, it fallbacks to using `--extension` flag in order to package an extension.

## Checklist

* [x] I have viewed, signed, and submitted the Contributor License Agreement.
* [ ] I have linked issue(s) that this PR should close using keywords or the Github UI (See [docs](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue))
* [x] I have added an integration test, if necessary.
* [x] I have reviewed the [styleguide](https://github.com/paketo-buildpacks/community/blob/main/STYLEGUIDE.md) for guidance on my code quality.
* [x] I'm happy with the commit history on this PR (I have rebased/squashed as needed).

### CODE REVIEW

**Comment by mhdawson** [comment]
Successfully used this in an integration test in https://github.com/nodeshift/ubi-nodejs-extension/pull/35

File: packagers/jam.go [hunk 1]
```diff
@@ -32,9 +33,21 @@ func (j Jam) WithExecutable(executable Executable) Jam {
 }
 
 func (j Jam) Execute(buildpackDir, output, version string, offline bool) error {
+
+	extensionTomlPath := filepath.Join(buildpackDir, "extension.toml")
+
+	buildpackOrExtensionToml := "buildpack.toml"
+	command := "--buildpack"
+	if _, err := os.Stat(extensionTomlPath); err == nil {
```
Comment (robdimsdale) [inline packagers/jam.go:41]: I'd probably suggest using `packit/fs.Exists` here, as I think it's easier to reason about.

File: packagers/jam.go [hunk 2]
```diff
@@ -32,9 +33,21 @@ func (j Jam) WithExecutable(executable Executable) Jam {
...
+		buildpackOrExtensionToml = "extension.toml"
+		command = "--extension"
+	}
+
+	fmt.Println("jam", command, filepath.Join(buildpackDir, buildpackOrExtensionToml), "--output", output, "--version", version, "--offline", offline)
```
Comment (robdimsdale) [inline packagers/jam.go:46]: I'd probably remove this println unless you have a compelling reason to keep it in. I'd prefer to err on the side of silence - especially in the happy-path. I don't think we have any good existing model for debug logging in occam, and if/when we create that, I'd like to apply it universally.

**Review by robdimsdale — COMMENTED** [review]
In principle I think this is fine, though I have a couple of minor changes I'd like to see. Once we've updated them I'd be happy to merge.

In general, I don't love the idea of silently selecting buildpack.toml vs extension.toml or silently falling back from one to the other, but I'm having a hard time coming up with a better interface at this time. So I'm happy with this general direction.

File: packagers/jam.go (hunk 2 above)
Comment (pacostas) [reply packagers/jam.go]: > In principle I think this is fine, though I have a couple of minor changes I'd like to see. Once we've updated them I'd be happy to merge.
> 
> In general, I don't love the idea of silently selecting buildpack.toml vs extension.toml or silently falling back from one to the other, but I'm having a hard time coming up with a better interface at this time. So I'm happy with this general direction.

I also dont agree with the idea of fallback, but for now looks like a good alternative as I tried adding an extra argument on the `func (j Jam) Execute(buildpackDir, output, version string, offline bool) error {` and it turned out with a lots of errors.

Added your suggestions, I guess next step before merging this pr is to wait for the https://github.com/paketo-buildpacks/jam/pull/234

**Comment by robdimsdale** [comment]
@pacostas looks good. Can you fix the lint error? (Comparing `fileExists == true` can be shortened to just `fileExists`). Then this PR will be good to once https://github.com/paketo-buildpacks/jam/pull/234 is merged.

**Review by robdimsdale — APPROVED** [review]

**Comment by ForestEckhardt** [comment]
https://github.com/paketo-buildpacks/jam/releases/tag/v2.3.0

**Pull request merged**